Update rubygem-foreman_remote_execution to 2.0.2

lzap opened the pull request:
For plugin updates, please indicate which repos this should be built into:

* [x] Nightly
* [x] 1.24

DO NOT MERGE YET

The GEM is actually not 2.0.0 but 2.0.1 scratchbuild, I want to test the
SELinux packaging changes end to end first.

lzap commented:
This PR breaks dependencies, how to handle that?

```
[2019-10-21T12:40:22.703Z]     package: tfm-rubygem-foreman_ansible-3.0.7-1.fm1_24.el7.noarch from el7-foreman-plugins-nightly
[2019-10-21T12:40:22.703Z]       unresolved deps:
[2019-10-21T12:40:22.703Z]          tfm-rubygem(foreman_remote_execution) < 0:2.0
[2019-10-21T12:40:22.703Z]     package: tfm-rubygem-foreman_salt-12.0.0-1.fm1_24.el7.noarch from el7-foreman-plugins-nightly
[2019-10-21T12:40:22.703Z]       unresolved deps:
[2019-10-21T12:40:22.703Z]          tfm-rubygem(foreman_remote_execution) < 0:1.9
```

ekohl commented:
Those will need releases to allow the new major version. Doesn't have to block the PR but it'd be great if the maintainer of those plugins prepared something. Then we can merge it all (even if via multiple PRs).

lzap commented:
Those are already pending review, filed two weeks ago.

Thanks for the requires, can I just blindly copy and paste them? Since I aleady have local changes, the script will not work for me anymore.

ekohl commented:
> Thanks for the requires, can I just blindly copy and paste them? Since I aleady have local changes, the script will not work for me anymore.

That's exactly why I pasted them.

ehelms commented on packages/plugins/rubygem-foreman_remote_execution/rubygem-foreman_remote_execution.spec:
```diff
@@ -9,42 +9,70 @@
+
+# start package.json dependencies BuildRequires
+BuildRequires: npm(@theforeman/vendor) >= 1.4.0
+BuildRequires: npm(@theforeman/vendor) < 2.0.0
```
All of the `npm(` calls will need updating to `%{?scl_prefix}npm(` due to the chnage over to using NodeJS SCL in our builds.

lzap replied:
Just like that?

lzap commented:
Rebased, upstream released 2.0.2 which actually removed the launcher-wrapper. So I removed it from the RPM as well - I haven't realized that Cockpit already has a policy so no wrapper needed.

https://github.com/theforeman/foreman_remote_execution/pull/441

lzap commented:
Hmmmm, not sure I understand.

```
[2019-10-29T14:48:25.879Z] fatal: [rubygem-foreman_remote_execution]: FAILED! => changed=true 

[2019-10-29T14:48:25.879Z]   ansible_facts:

[2019-10-29T14:48:25.879Z]     discovered_interpreter_python: /usr/bin/python

[2019-10-29T14:48:25.879Z]   cmd:

[2019-10-29T14:48:25.879Z]   - rpmlint

[2019-10-29T14:48:25.879Z]   - --file

[2019-10-29T14:48:25.879Z]   - /home/jenkins/workspace/foreman-packaging-rpm-pr-test/packages/plugins/rubygem-foreman_remote_execution/.rpmlintrc

[2019-10-29T14:48:25.879Z]   - /home/jenkins/workspace/foreman-packaging-rpm-pr-test/packages/plugins/rubygem-foreman_remote_execution

[2019-10-29T14:48:25.879Z]   delta: '0:00:00.546803'

[2019-10-29T14:48:25.879Z]   end: '2019-10-29 14:48:25.708197'

[2019-10-29T14:48:25.879Z]   msg: non-zero return code

[2019-10-29T14:48:25.879Z]   rc: 64

[2019-10-29T14:48:25.879Z]   start: '2019-10-29 14:48:25.161394'

[2019-10-29T14:48:25.879Z]   stderr: ''

[2019-10-29T14:48:25.879Z]   stderr_lines: <omitted>

[2019-10-29T14:48:25.879Z]   stdout: |-

[2019-10-29T14:48:25.879Z]     /home/jenkins/workspace/foreman-packaging-rpm-pr-test/packages/plugins/rubygem-foreman_remote_execution/rubygem-foreman_remote_execution.spec: W: no-%prep-section

[2019-10-29T14:48:25.879Z]     /home/jenkins/workspace/foreman-packaging-rpm-pr-test/packages/plugins/rubygem-foreman_remote_execution/rubygem-foreman_remote_execution.spec: W: no-%build-section

[2019-10-29T14:48:25.879Z]     /home/jenkins/workspace/foreman-packaging-rpm-pr-test/packages/plugins/rubygem-foreman_remote_execution/rubygem-foreman_remote_execution.spec: W: no-%install-section

[2019-10-29T14:48:25.879Z]     /home/jenkins/workspace/foreman-packaging-rpm-pr-test/packages/plugins/rubygem-foreman_remote_execution/rubygem-foreman_remote_execution.spec: E: specfile-error error: line 1: Unknown tag: /annex/objects/SHA256E-s14325--2b30e766d78445a3b190a933659f6340a978713d135b5b56638e865f4b2375f4.spec

[2019-10-29T14:48:25.879Z]     /home/jenkins/workspace/foreman-packaging-rpm-pr-test/packages/plugins/rubygem-foreman_remote_execution/rubygem-foreman_remote_execution.spec: E: specfile-error error: query of specfile /home/jenkins/workspace/foreman-packaging-rpm-pr-test/packages/plugins/rubygem-foreman_remote_execution/rubygem-foreman_remote_execution.spec failed, can't parse

[2019-10-29T14:48:25.879Z]     0 packages and 1 specfiles checked; 2 errors, 3 warnings.

[2019-10-29T14:48:25.879Z]   stdout_lines: <omitted>
```

ehelms commented:
The spec file was git annex added which essentially removed it and is the reason for the error.

lzap commented:
Ok switching to new branch did help: https://github.com/theforeman/foreman-packaging/pull/4247
